File: QE-Selenium-Java/Activity13.java
package demos;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity13 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/tables");

        System.out.println("Page Title: " + driver.getTitle());

        List<WebElement> rows = driver.findElements(
                By.xpath("//table/tbody/tr"));
        System.out.println("No of rows: " + rows.size());

        List<WebElement> columns = driver.findElements(
                By.xpath("//table/thead/tr/th"));
        System.out.println("No of columns: " + columns.size());

        List<WebElement> thirdRow = driver.findElements(
                By.xpath("//table/tbody/tr[3]/td"));

        System.out.println("Third row data:");
        for (WebElement cell : thirdRow) {
            System.out.println(cell.getText());
        }

        WebElement cellVal = driver.findElement(
                By.xpath("//table/tbody/tr[2]/td[2]"));
        System.out.println("2nd row, 2nd column data: " + cellVal.getText());

        driver.quit();
    }
}

File: QE-Selenium-Python/Activity13.py
from selenium import webdriver
from selenium.webdriver.common.by import By

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/tables")

print("Page Title:", driver.title)

rows = driver.find_elements(By.XPATH, "//table/tbody/tr")
print("No of rows:", len(rows))

colunms = driver.find_elements(By.XPATH, "//table/thead/tr/th")
print("No of columns:", len(colunms))

third_row = driver.find_elements(By.XPATH, "//table/tbody/tr[3]/td")

print("Third row data:")
for cell in third_row:
    print(cell.text)

cell_val = driver.find_element(By.XPATH, "//table/tbody/tr[2]/td[2]")
print("2nd row, 2nd column data:", cell_val.text)

driver.quit()